runtime/include/chpl-tasks.h

```diff
@@ -228,6 +228,20 @@ c_sublocid_t chpl_task_getRequestedSubloc(void);
 chpl_taskID_t chpl_task_getId(void);
 
 //
+// Checks whether two task IDs are the same
+//
+chpl_bool chpl_task_idEquals(chpl_taskID_t, chpl_taskID_t);
+
+//
+// Returns the string representation of task ID
+// The string returned is the same buffer passed as argument
```

Comment:
I returned the buffer, so that the tostring() function can directly be used for printing. Somewhat similar to how tostring() is done in java and c#. 
For example, 
printf(“task id %s used %d cycles\n”, chpl_task_idToString(buff, 10, ID), num_cycle);

Otherwise, we will have to invoke the tostring() function in a separate call and then use the buffer in printf.

One solution is that the implementation of chpl_task_idToString should check whether the buffer was big enough and the conversion is successful. If not, print a warning (chapel_warning()) and proceed. I don't think any error in toString() should cause the program to exit.